helix-term/src/commands/typed.rs

```diff
@@ -1801,22 +1800,48 @@ fn toggle_option(
     let pointer = format!("/{}", key.replace('.', "/"));
     let value = config.pointer_mut(&pointer).ok_or_else(key_error)?;
 
-    let Value::Bool(old_value) = *value else {
-        anyhow::bail!("Key `{}` is not toggle-able", key)
+    *value = match value.as_bool() {
+        Some(value) => serde_json::Value::Bool(!value),
+        None => {
+            if args.len() != 2 {
+                anyhow::bail!(
+                    "Bad arguments. For non-boolean configurations use: `:toggle key val1,val2`"
+                );
+            }
+            if !value.is_string() {
+                anyhow::bail!("Bad configuration. Cannot cycle non-string configurations");
+            }
+
+            // The comma separated values to cycle
+            let first_arg = &args[1];
+            let field_error = |_| anyhow::anyhow!("Could not parse field `{}`", first_arg);
+
+            first_arg
+                .split_once(',')
+                .map(|(first, remaining)| {
+                    Ok(serde_json::Value::String(
+                        [first]
+                            .into_iter()
+                            .chain(remaining.split(','))
+                            .skip_while(|e| e != value)
+                            .nth(1)
+                            .unwrap_or(first)
+                            .to_string(),
+                    ))
+                })
+                .unwrap_or_else(|| first_arg.parse().map_err(field_error))?
```

Comment:
this feels odd. Instead of doing the parsing yourself why not simply require one argument per value?

```
:toggle text-width 10 80 100
```

the check above would turn into `ensure(args.len()  > 2,...)`

This feels more natural to me and works better with the existing infrastructure around commands

Reply:
I don't know what I was thinking. Much cleaner, see https://github.com/helix-editor/helix/pull/4411/commits/998426efcf754b738914a9d92b04e1a6a5bca5aa